### PR TITLE
[Chore] Update jsonnet manifest create query frontend discovery only when it is necessary

### LINKED ISSUE #4259: Don't set query-frontend address on queriers when we run query-scheduler.
This issue is about our jsonnet manifests in `operations/mimir`.

When query-scheduler is enabled, or we don't run in the microservices mode:
https://github.com/grafana/mimir/blob/c87020a36be7c6ba483e9bc7e17f2ee956b29131/operations/mimir/query-scheduler.libsonnet#L44-L44

There's no need for queriers to find query-frontends, as they talk to query-scheduler, or mimir-backend in read-write mode.

So this option doesn't need to be set:
https://github.com/grafana/mimir/blob/c87020a36be7c6ba483e9bc7e17f2ee956b29131/operations/mimir/querier.libsonnet#L24-L24

And the query-frontend-discovery service doesn't have to be created:

https://github.com/grafana/mimir/blob/c87020a36be7c6ba483e9bc7e17f2ee956b29131/operations/mimir/query-frontend.libsonnet#L57-L62

### PR DESCRIPTION
**Pull request by ying-jeanne**
[Jsonnet] Update jsonnet manifest create query frontend discovery only when it is necessary

#### What this PR does
Update jsonnet manifest, avoid setting querier.frontend-address or create query-frontend-discovery when deployment mode is microserivces or query-scheduler is enabled. 

#### Which issue(s) this PR fixes or relates to

Fixes #4259

#### Checklist

- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pracucci — COMMENTED** [review]
The changes LGTM. Please add a `[CHANGE]` CHANGELOG entry under "### Jsonnet", and switch to "Ready to review" once you feel comfortable?

File: CHANGELOG.md [hunk 1]
```diff
@@ -80,6 +80,7 @@ Querying with using `{__mimir_storage__="ephemeral"}` selector no longer works.
 
 ### Jsonnet
 
+* [CHANGE] Create the `query-frontend-discovery` service only when Miimr is deployed in microservice mode without query-scheduler. #4353
```
Comment (pracucci) [inline CHANGELOG.md:83]:
```suggestion
* [CHANGE] Create the `query-frontend-discovery` service only when Mimir is deployed in microservice mode without query-scheduler. #4353
```

**Review by pracucci — APPROVED** [review]
LGTM!

**Review by pstibrany — APPROVED** [review]
Thank you, also thanks for taking care of Helm chart!

**Pull request merged**